src/internal/header.rs

```diff
@@ -74,6 +77,20 @@ impl Header {
 
     /// Convert this `Header` into a `Package`
     pub(crate) fn to_package(&self) -> Package {
+        let mut filenames = Vec::new();
+        // Always safe with a valid td (self.0) and fi:
+        unsafe {
+            let fi = librpm_sys::rpmfiNew(null_mut(), self.0, 0, 0);
+            while librpm_sys::rpmfiNext(fi) >= 0 {
+                let name = CStr::from_ptr(librpm_sys::rpmfiFN(fi))
```

Comment:
It's a little more subtle than that. Maybe something like this for the docs?

```suggestion
        // SAFETY: see inline notes about how memory is managed
        unsafe {
            // SAFETY: `fi` is valid while being iterated over
            let fi = librpm_sys::rpmfiNew(null_mut(), self.0, 0, 0);
            while librpm_sys::rpmfiNext(fi) >= 0 {
                // SAFETY: `name` is valid until the next invocation of `rpmfiNext`
                let name = CStr::from_ptr(librpm_sys::rpmfiFN(fi))
```

Reply:
I have attempted to encode it a bit further in safe abstractions and add (private for now) iterator wrapping fi. Also fixed memory leak from not calling `rpmfiFree`  and removed useless intermediate allocation while creating PathBuf.